### PR TITLE
Issue #PS-657 fix:ALL screen UI matach and Bug fixes

### PR DESCRIPTION
**Pull request by shreyas1434shinde**
## Description

Please describe the changes in your pull request in few words here.

### Changes

List the changes done to fix a bug or introducing a new feature.

## How to test

Describe the steps required to test the changes proposed in the pull request.

Please consider using the closing keyword if the pull request is proposed to
fix an issue already created in the repository
(https://help.github.com/articles/closing-issues-using-keywords/)

## Summary by CodeRabbit

- **New Features**
  - Enhanced date formatting and conditional checks in the Dashboard.
  - Improved cohort selection dropdown and attendance status messaging.

- **Style**
  - Updated styling for buttons, typography, and layout elements across various components.
  - Adjusted z-index and background color for modal and main elements.

- **Bug Fixes**
  - Fixed issues with UI element disabling based on `classId`.
  - Corrected comparison operators for string values.

- **Refactor**
  - Reorganized import statements and improved component structure for better readability and maintainability.

### CODE REVIEW

File: src/components/LearnerModal.tsx [hunk 1]
```diff
@@ -95,84 +99,84 @@ const LearnerModal = ({
...
                         margin={0}
-                        color={'#969088'}
+                        // variant="h6"
+                        lineHeight={'16px'}
+                        fontSize={'12px'}
+                        fontWeight={'600'}
+                        color={theme.palette.warning['500']}
                       >
-                        {userName}
+                        Full Name
```
Comment (itsvick) [inline src/components/LearnerModal.tsx:123]: @shreyas1434shinde remove hard coded value

File: src/components/LearnerModal.tsx [hunk 2]
```diff
@@ -95,84 +99,84 @@ const LearnerModal = ({
...
                           fontWeight={'400'}
                           lineHeight={'24px'}
                           margin={0}
-                          color={'#969088'}
-                          style={{
-                            wordBreak: 'break-word',
-                            whiteSpace: 'normal',
-                          }}
-                          // noWrap
+                          color={'#4D4639'}
```
Comment (itsvick) [inline src/components/LearnerModal.tsx:131]: @shreyas1434shinde get color from the theme

File: src/components/UpDownButton.tsx [hunk 3]
```diff
@@ -17,10 +17,10 @@ const UpDownButton = () => {
     const viewportHeight = document.documentElement.clientHeight;
 
     const atBottom =
-      pathname === '/attendance-overview' ? scrolled >= 300 : scrolled >= 720;
+      pathname === '/attendance-overview' ? scrolled >= 294 : scrolled >= 700;
```
Comment (itsvick) [inline src/components/UpDownButton.tsx:20]: @shreyas1434shinde have u tested this on multiple devices?

Comment (shreyas1434shinde) [reply src/components/UpDownButton.tsx:20]: @itsvick yes

**Pull request merged**